08_analysis_tools/CalibrateHallSensor.m

```diff
@@ -0,0 +1,111 @@
+% / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / /
+% Deterime hall sensor calibration curve
+% / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / /
+
+% !!-- This script must be run *after running the logconv.m script. --!!
+
+% This script fits a 3rd order polynomial describing the relation between
+% magnetic field strength (mT) measured by the hall sensor to the angular
+% offset from the boom (deg). Calibration parameters are output in the
+% suitable form for use with PX4 firmware.
+
+% INSTRUCTIONS:
+% 1) fix vane boom to calibration dial
+% 2) connect hall sensor to pixhawk / boot pixhawk
+% 3) choose a starting/ending angle (note them down: angle_st, angle_ed)
+% 4) move the vane such that the tip of the boom is centered on the
+% starting angle (prior to this, it is somewhat helpful to make some rapid
+% / large motions with the vane such that it is easier to find the
+% calibration starting point in the log data)
+% 5) hold the vane still at the starting angle for several seconds, then
+% move on directly to the next angles, pausing at each for several seconds,
+% until reaching the end angle.
+% NOTE: it is advised to take steps of 5 degrees at larger angles, until
+% reaching e.g. +/- 5-10 degrees, where the step size shold be reduced
+% (e.g. 1 or 2 degrees) in an attempt to give a higher accuracy fit at the
+% more common values the vane will take while in flight.
```

Comment:
Maybe add here which params correspond to which limit, other than that the instructions are very clear

Reply:
I made a small modification to this note about what calibration option to set the small range with -- were you referring to this, or something else?

Reply:
Yes, now it is clearer